extensions/iam/oauth2/src/main/java/org/eclipse/dataspaceconnector/iam/oauth2/Oauth2Extension.java

```diff
@@ -111,21 +134,40 @@ public void initialize(ServiceExtensionContext context) {
         CertificateResolver certificateResolver = context.getService(CertificateResolver.class);
         configBuilder.certificateResolver(certificateResolver);
 
-        configBuilder.objectMapper(context.getTypeManager().getMapper());
-
-        Oauth2Configuration configuration = configBuilder.build();
-
-        // for now, lets assume we have RSA Private keys
-        Supplier<JWSSigner> pkSuppplier = () -> {
-            var pkId = configuration.getPrivateKeyAlias();
-            var pk = configuration.getPrivateKeyResolver().resolvePrivateKey(pkId, RSAPrivateKey.class);
-            return pk == null ? null : new RSASSASigner(pk);
-        };
-        IdentityService oauth2Service = new Oauth2ServiceImpl(configuration, pkSuppplier);
+        return configBuilder.build();
+    }
 
-        context.registerService(IdentityService.class, oauth2Service);
+    private OkHttpClient createClient() {
+        try {
+            KeyStore keyStore = readKeyStore();
+            SSLContext sslContext = SSLContext.getInstance("SSL");
+            TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+            KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+
+            trustManagerFactory.init(keyStore);
+            keyManagerFactory.init(keyStore, "test123".toCharArray());
+            sslContext.init(keyManagerFactory.getKeyManagers(), trustManagerFactory.getTrustManagers(), new SecureRandom());
+
+            return new OkHttpClient.Builder()
```

Comment:
I think we should avoid to instantiate a new `OkHttpClient`, is better to create a new one starting from the one already registered using the `newBuilder()` method, so you don't have to redefine basic configurations (like `connectTimeout` and `readTimeout`)

Reply:
corrected